core/payment/src/api/allocations.rs

```diff
@@ -162,12 +163,83 @@ async fn get_allocation(
     }
 }
 
+fn amend_allocation_fields(
+    old_allocation: Allocation,
+    new_allocation: NewAllocation,
+) -> Allocation {
+    Allocation {
+        allocation_id: old_allocation.allocation_id.clone(),
+        address: new_allocation
+            .address
+            .unwrap_or_else(|| old_allocation.address.clone()),
+        payment_platform: new_allocation
+            .payment_platform
+            .unwrap_or_else(|| old_allocation.payment_platform.clone()),
```

Comment:
I don't think updating platform makes any sense.
We should return error in such case.

Updating platform can cause unwanted problems if you already spent/reserved money on different platform

Reply:
Done